django/contrib/auth/management/__init__.py

```diff
@@ -141,46 +147,108 @@ def rename_permissions_after_model_rename(
     if not renames:
         return
 
-    for old_name, new_name in renames:
-        old_suffix = "_" + old_name.lower()
-        new_suffix = "_" + new_name.lower()
-
+    stdout = kwargs.get("stdout", sys.stdout)
+    conflicts = []
+    for app_label, from_model_name, to_model_name in renames:
+        try:
+            new_model = apps.get_model(app_label + "." + to_model_name)
+        except LookupError:
+            continue
         perms = Permission.objects.using(using).filter(
             content_type__app_label=app_label,
-            codename__endswith=old_suffix,
+            codename__in=[
+                action + "_" + from_model_name.lower()
+                for action in new_model._meta.default_permissions
+            ],
         )
 
         for perm in perms:
-            for action in ("add", "change", "delete", "view"):
+            for action in new_model._meta.default_permissions:
                 if not perm.codename.startswith(action + "_"):
                     continue
-
-                new_codename = f"{action}{new_suffix}"
-                new_name_str = f"Can {action} {camel_case_to_spaces(new_name)}"
-
+                new_codename, new_name_str = _get_permission_for_action(
+                    action, new_model._meta
+                )
                 if perm.codename == new_codename and perm.name == new_name_str:
                     continue
 
+                old_name_str = perm.name
                 perm.codename = new_codename
                 perm.name = new_name_str
 
                 try:
                     perm.save(update_fields={"codename", "name"}, using=using)
                 except IntegrityError:
-                    # IntegrityError is most likely caused by duplicate
-                    # permissions created in pre-Django 6.0 projects before
-                    # permission renames were handled.
+                    # Conflicts are most likely to arise from pre-Django 6.0
+                    # projects before permission renames were handled.
```

Comment:
You were probably right after all to avoid mentioning this -- on second (third?) thought, it's a little presumptuous.